### PR TITLE
feat: changes in SIWE signature info section

### LINKED ISSUE #24649: Redesign Signature SIWE: create info section
Screenshot 2024-05-20 at 9 31 06 PM

Figma: https://www.figma.com/design/v8npDTFD2fofLjDfbJUD9j/Confirmation-redesign-V3?node-id=360%3A4051&t=EuGBZwCijC3AjXa8-1

### PR DESCRIPTION
**Pull request by jpuri**
## **Description**

Change in info section for SIWE signature type.

## **Related issues**

Fixes: #24649

## **Manual testing steps**
1. Go to test dapp
2. Submit SIWE signature type
3. Check info section

## **Screenshots/Recordings**
Screenshot 2024-05-28 at 2 01 03 PM

## **Pre-merge author checklist**

- [X] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [X] I've completed the PR template to the best of my ability
- [X] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [X] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by digiwand** [comment]
^^ will need to resolve conflicts

May we add details to what the changes were in the description please?

File: ui/pages/confirmations/components/confirm/info/personal-sign/personal-sign.tsx [hunk 1]
```diff
@@ -29,6 +31,10 @@ const PersonalSignInfo: React.FC = () => {
     return null;
   }
 
+  const { from } = currentConfirmation.msgParams;
+
+  console.log('=======', JSON.stringify(currentConfirmation));
+
```
Comment (digiwand) [inline ui/pages/confirmations/components/confirm/info/personal-sign/personal-sign.tsx:37]:
```suggestion
```

File: ui/pages/confirmations/hooks/useCurrentConfirmation.ts [hunk 2]
```diff
@@ -76,6 +76,8 @@ const useCurrentConfirmation = () => {
         return;
       }
 
+      // comment if condition below to enable re-design for SIWE signatures
+      // this can be removed once SIWE code changes are completed
```
Comment (digiwand) [inline ui/pages/confirmations/hooks/useCurrentConfirmation.ts:80]: I'm confused as to what the purpose of this comment is. could you elaborate/re-elaborate please?

File: ui/pages/confirmations/utils/confirm.ts [hunk 3]
```diff
@@ -23,3 +26,12 @@ const SIGNATURE_APPROVAL_TYPES = [
...
 ) => SIGNATURE_APPROVAL_TYPES.includes(request.type as ApprovalType);
+
+export const parseTypedDataMessage = (dataToParse: string) => {
+  const { message, domain = {}, primaryType, types } = JSON.parse(dataToParse);
+  const sanitizedMessage = sanitizeMessage(message, primaryType, types);
+  return { domain, sanitizedMessage, primaryType };
+};
+
+export const isSIWESignatureRequest = (request: SignatureRequestType) =>
+  request.msgParams?.siwe?.isSIWEMessage;
```
Comment (digiwand) [inline ui/pages/confirmations/utils/confirm.ts:37]: [nit] would be nice to alphabetize to keep file organized

**Review by digiwand — DISMISSED** [review]
no blockers - added a couple of suggestions

File: ui/pages/confirmations/hooks/useCurrentConfirmation.ts (hunk 2 above)
Comment (jpuri) [reply ui/pages/confirmations/hooks/useCurrentConfirmation.ts:80]: This is added to help in further work on SIWE signature. Commenting out this if condition will result in SIWE signatures shown in new designs.

**Review by pedronfigueiredo — APPROVED** [review]

File: ui/pages/confirmations/hooks/useCurrentConfirmation.ts (hunk 2 above)
Comment (digiwand) [reply ui/pages/confirmations/hooks/useCurrentConfirmation.ts:80]: 👍🏼 
```suggestion
      // comment out if block below to enable redesigned SIWE signatures
      // TODO: remove once redesigned SIWE code changes are completed
```

**Review by digiwand — APPROVED** [review]

**Pull request merged**